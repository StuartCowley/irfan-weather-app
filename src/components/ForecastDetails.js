import React from "react";
import PropTypes from "prop-types";

function ForecastDetails({ forecast }) {
  const { date, temperature, wind, humidity } = forecast;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">Date: {formattedDate}</div>
      <div className="forecast-details__maxTemperature">
        Max Temp: {temperature.max}&deg;c
      </div>
      <div className="forecast-details__minTemperature">
        Min Temp: {temperature.min}&deg;c
      </div>
      <div className="forecast-details__windSpeed">
        Wind Speed: {wind.speed}mph
      </div>
      <div className="forecast-details__windDirection">
        Wind Direction: {wind.direction}
      </div>
      <div className="forecast-details__humidity">Humidity: {humidity}%</div>
    </div>
  );
}

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }),
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }),
    humidity: PropTypes.number,
  }).isRequired,
};
