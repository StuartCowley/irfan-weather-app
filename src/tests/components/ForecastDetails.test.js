import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1525046400000,
    temperature: {
      min: 4,
      max: 11,
    },
    wind: {
      speed: 13,
      direction: "s",
    },
    humidity: 30,
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText("Date: Mon Apr 30 2018")).toHaveAttribute(
      "class",
      "forecast-details__date"
    );
    expect(getByText("Min Temp: 4°c")).toHaveAttribute(
      "class",
      "forecast-details__minTemperature"
    );
    expect(getByText("Max Temp: 11°c")).toHaveAttribute(
      "class",
      "forecast-details__maxTemperature"
    );
    expect(getByText("Wind Speed: 13mph")).toHaveAttribute(
      "class",
      "forecast-details__windSpeed"
    );
    expect(getByText("Wind Direction: s")).toHaveAttribute(
      "class",
      "forecast-details__windDirection"
    );
    expect(getByText("Humidity: 30%")).toHaveAttribute(
      "class",
      "forecast-details__humidity"
    );
  });
});
