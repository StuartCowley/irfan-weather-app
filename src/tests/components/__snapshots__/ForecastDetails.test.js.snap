// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Date: Mon Apr 30 2018
    </div>
    <div
      class="forecast-details__maxTemperature"
    >
      Max Temp: 11°c
    </div>
    <div
      class="forecast-details__minTemperature"
    >
      Min Temp: 4°c
    </div>
    <div
      class="forecast-details__windSpeed"
    >
      Wind Speed: 13mph
    </div>
    <div
      class="forecast-details__windDirection"
    >
      Wind Direction: s
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 30%
    </div>
  </div>
</DocumentFragment>
`;
